src/poetry/installation/executor.py

```diff
@@ -792,8 +793,19 @@ def _populate_hashes_dict(self, archive: Path, package: Package) -> None:
 
     @staticmethod
     def _validate_archive_hash(archive: Path, package: Package) -> str:
-        archive_hash: str = "sha256:" + get_file_hash(archive)
-        known_hashes = {f["hash"] for f in package.files if f["file"] == archive.name}
+        known_hashes: set[str] = {
+            f["hash"] for f in package.files if f["file"] == archive.name
+        }
+        hash_types: set[str] = {t.split(":")[0] for t in known_hashes}
+        hash_type: str | None = get_highest_priority_hash_type(hash_types, archive.name)
```

Comment:
```suggestion
        known_hashes = {f["hash"] for f in package.files if f["file"] == archive.name}
        hash_types = {t.split(":")[0] for t in known_hashes}
        hash_type = get_highest_priority_hash_type(hash_types, archive.name)
```

We don't need these annotations since the type is clear.